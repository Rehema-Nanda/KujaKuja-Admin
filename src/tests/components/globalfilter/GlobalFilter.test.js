import React from "react";
import { shallow } from "enzyme";
import moment from "moment";
import GlobalFilter from "../../../components/globalfilter/GlobalFilter";

describe("<GlobalFilter />", () => {
    const fixedDate = moment.utc("2019-08-08").startOf("day");
    const testProps = {
        dateStart: fixedDate.clone().subtract(7, "days"),
        dateEnd: fixedDate,
        allCountries: [],
        selectedCountries: [],
        allLocations: [],
        selectedLocations: [],
        selectedServiceTypes: [],
        selectedUsers: [],
        selectedServicePoints: [],
        selectedSurveyTypes: [],
        selectedSatisfied: [],
        selectedFreshIdea: [],
        users: [{id: "1"}],
        locations: [{id: "1"}],
        countries: [{id: "1"}],
        booleanOptions: [{id: "1"}],
        surveyTypes: [{id: "1"}],
        servicePoints: [{id: "1"}],
    };

    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<GlobalFilter {...testProps} />);
    });

    it("should render correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should set mobile filter true when globalfilter back button is clicked", () => {
        wrapper.setState({showMobileFilter: false});
        const div = wrapper.find("div.globalfilter-back-button");
        div.simulate("click");
        expect(wrapper.state().showMobileFilter).toBe(true);
    });

    it("should change the state of mobile controls when toggleMobileControlsAndClose function is called", () => {
        navigator.__defineGetter__("userAgent", function() {
            return "Chrome/75.0.3770.142 Safari/537.3";
        });
        const event = {
            target: {
                style: {},
            },
        };
        const toggle = wrapper.find("NavbarToggler");
        toggle.simulate("click", event);
        expect(wrapper.state().showMobileFilter).toBe(true);
    });

    it("should set image sizes when mobileControlsOpen state is false and toggleMobileControlsAndClose function is called", () => {
        wrapper.setState({mobileControlsOpen: true});
        navigator.__defineGetter__("userAgent", function(){
            return "Chrome/75.0.3770.142 Safari/537.3";
        });
        const event = {
            target: {
                style: {},
            },
        };
        const toggle = wrapper.find("NavbarToggler");
        toggle.simulate("click", event);
        expect(event.target.style).toBeTruthy();
    });

    it("should change the state of showMobileFilterCountries when displayMobileFilterCountries function is called", () => {
        const button = wrapper.find("GlobalFilterMobileButton").at(3);
        button.simulate("click");
        expect(wrapper.state().showMobileFilterCountries).toBe(true);
    });

    it("should change the state of showMobileFilterLocations when displayMobileFilterLocations function is called", () => {
        const button = wrapper.find("GlobalFilterMobileButton").at(4);
        button.simulate("click");
        expect(wrapper.state().showMobileFilterLocations).toBe(true);
    });

    it("should change the state of showMobileFilterUsers when displayMobileFilterUsers function is called", () => {
        const button = wrapper.find("GlobalFilterMobileButton").first();
        button.simulate("click");
        expect(wrapper.state().showMobileFilterUsers).toBe(true);
    });

    it("should change the state of showMobileFilterServicePoints when displayMobileFilterServicePoints function is called", () => {
        const button = wrapper.find("GlobalFilterMobileButton").at(5);
        button.simulate("click");
        expect(wrapper.state().showMobileFilterServicePoints).toBe(true);
    });

    it("should change the state of showMobileFilterStartDate when displayMobileFilterStartDate function is called", () => {
        const button = wrapper.find("GlobalFilterMobileButton").at(1);
        button.simulate("click");
        expect(wrapper.state().showMobileFilterStartDate).toBe(true);
    });

    it("should change the state of showMobileFilterEndDate when displayMobileFilterEndDate function is called", () => {
        const button = wrapper.find("GlobalFilterMobileButton").at(2);
        button.simulate("click");
        expect(wrapper.state().showMobileFilterEndDate).toBe(true);
    });

    it("should change the state of showMobileFilterSurveyTypes when displayMobileFilterSurveyTypes function is called", () => {
        const button = wrapper.find("GlobalFilterMobileButton").at(6);
        button.simulate("click");
        expect(wrapper.state().showMobileFilterSurveyTypes).toBe(true);
    });

    it("should change the state of showMobileFilterSatisfied when displayMobileFilterSatisfied function is called", () => {
        const button = wrapper.find("GlobalFilterMobileButton").at(7);
        button.simulate("click");
        expect(wrapper.state().showMobileFilterSatisfied).toBe(true);
    });

    it("should change the state of showMobileFilterFreshIdea when displayMobileFilterFreshIdea function is called", () => {
        const button = wrapper.find("GlobalFilterMobileButton").at(8);
        button.simulate("click");
        expect(wrapper.state().showMobileFilterFreshIdea).toBe(true);
    });
});
