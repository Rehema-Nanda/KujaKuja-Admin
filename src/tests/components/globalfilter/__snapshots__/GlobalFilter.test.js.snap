// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalFilter /> should render correctly 1`] = `
"<div>
  <div className=\\"globalfilter-container \\">
    <div className=\\"globalfilter-desktop\\">
      <div className=\\"globalfilter-search-container\\">
        <GlobalFilterSearch setData={[undefined]} toggleRefresh={[undefined]} placeholder=\\"Idea Search\\" clearSearch={false} setClearSearchFalse={[undefined]} defaultSearchValue=\\"\\" />
      </div>
      <div className=\\"globalfilter-locations-container\\">
        <GlobalFilterDropdown id=\\"CountriesDropDown\\" title=\\"Countries\\" items={{...}} itemLabelProperty=\\"name\\" selectItemHandler={[undefined]} selectItemHandlerArgs={{...}} selectedItems={{...}} showSearch={true} isServicePoint={false} />
        <div className=\\"globalfilter-locations-spacer\\" />
        <GlobalFilterDropdown id=\\"LocationsDropDown\\" className=\\"location-dropdown\\" title=\\"Locations\\" items={{...}} itemLabelProperty=\\"name\\" selectItemHandler={[undefined]} selectItemHandlerArgs={{...}} selectedItems={{...}} showSearch={true} isServicePoint={false} />
      </div>
      <div className=\\"globalfilter-calendar-container\\">
        <GlobalFilterCalendar title=\\"From:\\" id=\\"CalendarDropDown\\" withPortal={false} setDateStartEnd={[undefined]} type=\\"calendar_date\\" dateStart={{...}} dateEnd={{...}} />
      </div>
      <div className=\\"globalfilter-service-container\\">
        <GlobalFilterServices serviceTypes={[undefined]} toggleServiceType={[undefined]} selectedServiceTypes={{...}} />
      </div>
    </div>
    <div className=\\"globalfilter-desktop\\">
      <div className=\\"globalfilter-users-container\\">
        <GlobalFilterDropdown id=\\"UsersDropDown\\" title=\\"Users\\" items={{...}} itemLabelProperty=\\"email\\" selectItemHandler={[undefined]} selectItemHandlerArgs={{...}} selectedItems={{...}} showSearch={true} isServicePoint={false} />
      </div>
      <div className=\\"globalfilter-service-points-container\\">
        <GlobalFilterDropdown id=\\"ServicePointsDropDown\\" title=\\"Service Points\\" items={{...}} itemLabelProperty=\\"name\\" selectItemHandler={[undefined]} selectItemHandlerArgs={{...}} selectedItems={{...}} showSearch={true} isServicePoint={false} />
      </div>
      <div className=\\"globalfilter-survey-types-container\\">
        <GlobalFilterDropdown id=\\"SurveyTypeDropDown\\" title=\\"Survey Types\\" items={{...}} itemLabelProperty=\\"name\\" selectItemHandler={[undefined]} selectItemHandlerArgs={{...}} selectedItems={{...}} isServicePoint={false} showSearch={false} />
      </div>
      <div className=\\"globalfilter-satisfied-container\\">
        <GlobalFilterDropdown id=\\"SatisfiedDropDown\\" title=\\"Satisfied\\" items={{...}} itemLabelProperty=\\"name\\" selectItemHandler={[undefined]} selectItemHandlerArgs={{...}} selectedItems={{...}} isServicePoint={false} showSearch={false} />
      </div>
      <div className=\\"globalfilter-fresh-idea-container\\">
        <GlobalFilterDropdown id=\\"FreshIdeaDropDown\\" title=\\"Fresh Idea\\" items={{...}} itemLabelProperty=\\"name\\" selectItemHandler={[undefined]} selectItemHandlerArgs={{...}} selectedItems={{...}} isServicePoint={false} showSearch={false} />
      </div>
      <div className=\\"globalfilter-reset-filters-container\\">
        <Button className=\\"resetButton primary\\" onClick={[undefined]} color=\\"secondary\\" tag=\\"button\\">
          Reset All
        </Button>
      </div>
    </div>
    <div className=\\"globalfilter-mobile\\">
      <Navbar tag=\\"nav\\" expand={false}>
        <NavbarBrand className=\\"globalfilter-mobile-brand\\" tag=\\"a\\">
          <p>
            Filter
          </p>
        </NavbarBrand>
        <NavbarToggler onClick={[Function: bound ]} tag=\\"button\\" type=\\"button\\" />
        <Collapse isOpen={false} navbar={true} in={false} mountOnEnter={false} unmountOnExit={false} appear={false} enter={true} exit={true} onEnter={[Function: noop]} onEntering={[Function: noop]} onEntered={[Function: noop]} onExit={[Function: noop]} onExiting={[Function: noop]} onExited={[Function: noop]} tag=\\"div\\" timeout={350}>
          <div className=\\"globalfilter-mobile-container\\">
            <div>
              <div className=\\"globalfilter-search-container fadein\\">
                <GlobalFilterSearch setData={[undefined]} toggleRefresh={[undefined]} placeholder=\\"Idea Search\\" clearSearch={false} defaultSearchValue=\\"\\" />
              </div>
              <div className=\\"globalfilter-users-container fadein\\">
                <GlobalFilterMobileButton text=\\"Users (0)\\" onClick={[Function: bound ]} />
              </div>
              <div className=\\"globalfilter-label fadein\\">
                <p className=\\"small\\">
                  Start Date:
                </p>
              </div>
              <div className=\\"globalfilter-label fadein\\">
                <p className=\\"small\\">
                  End Date:
                </p>
              </div>
              <div className=\\"globalfilter-calendar-container fadein\\">
                <GlobalFilterMobileButton text=\\"08/01/2019\\" onClick={[Function: bound ]} />
                <GlobalFilterMobileButton text=\\"08/08/2019\\" onClick={[Function: bound ]} />
              </div>
              <div className=\\"globalfilter-locations-container fadein\\">
                <GlobalFilterMobileButton text=\\"Countries (0)\\" onClick={[Function: bound ]} />
                <GlobalFilterMobileButton text=\\"Locations (0)\\" onClick={[Function: bound ]} />
              </div>
              <div className=\\"globalfilter-service-points-container fadein\\">
                <GlobalFilterMobileButton text=\\"Service Points (0)\\" onClick={[Function: bound ]} />
              </div>
              <div className=\\"globalfilter-service-container fadein\\">
                <GlobalFilterServices serviceTypes={[undefined]} toggleServiceType={[undefined]} selectedServiceTypes={{...}} />
              </div>
              <div className=\\"globalfilter-survey-types-container fadein\\">
                <GlobalFilterMobileButton text=\\"Survey Types (0)\\" onClick={[Function: bound ]} />
              </div>
              <div className=\\"globalfilter-satisfied-container fadein\\">
                <GlobalFilterMobileButton text=\\"Satisfied (0)\\" onClick={[Function: bound ]} />
              </div>
              <div className=\\"globalfilter-fresh-idea-container fadein\\">
                <GlobalFilterMobileButton text=\\"Fresh Idea (0)\\" onClick={[Function: bound ]} />
              </div>
              <div className=\\"globalfilter-reset-filters-container\\">
                <Button className=\\"resetButton primary\\" onClick={[undefined]} color=\\"secondary\\" tag=\\"button\\">
                  Reset All
                </Button>
              </div>
            </div>
          </div>
        </Collapse>
      </Navbar>
    </div>
  </div>
</div>"
`;
